src/main/java/hudson/security/LDAPSecurityRealm.java

```diff
@@ -813,45 +848,250 @@ public String getName() {
+        public boolean isCredentialsNonExpired() {
+            return userDetails.isCredentialsNonExpired();
+        }
+
+        @Override
+        public boolean isEnabled() {
+            return userDetails.isEnabled();
+        }
+
+        public LdapUserDetails getUserDetails() {
+            return userDetails;
+        }
+
+        public String getServer() {
+            return server;
+        }
+    }
+
+    private static class DelegateLDAPUserDetailsService implements UserDetailsService {
+        private final List<LDAPUserDetailsService> delegates;
+
+        public DelegateLDAPUserDetailsService() {
+            delegates = new ArrayList<>();
+        }
+
+        public void addDelegate(LDAPUserDetailsService delegate) {
+            delegates.add(delegate);
+        }
+
+        public boolean contains(LDAPUserDetailsService delegate) {
+            return delegates.contains(delegate);
+        }
+
+        @Override
+        public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException, DataAccessException {
+            UsernameNotFoundException lastUNFE = null;
+            for (LDAPUserDetailsService delegate : delegates) {
+                try {
+                    LdapUserDetails userDetails = delegate.loadUserByUsername(username);
+                    if (userDetails instanceof DelegatedLdapUserDetails) {
+                        return userDetails;
+                    } else {
+                        return new DelegatedLdapUserDetails(userDetails, delegate.server);
+                    }
+                } catch (UsernameNotFoundException e) {
+                    lastUNFE = e;
+                } catch (DataAccessException e) {
+                    LOGGER.log(Level.WARNING, "LDAP connection " + delegate.server + " seems to be broken, will _not_ try the next configuration.", e);
+                    throw e;
+                }
+            }
+            if (lastUNFE != null) {
+                throw  lastUNFE;
+            } else {
+                throw new UserMayOrMayNotExistException("This is not intentional.", username);
+            }
         }
     }
 
     public static class LDAPUserDetailsService implements UserDetailsService {
         public final LdapUserSearch ldapSearch;
         public final LdapAuthoritiesPopulator authoritiesPopulator;
         public final LDAPGroupMembershipStrategy groupMembershipStrategy;
+        public final String server;
         /**
          * {@link BasicAttributes} in LDAP tend to be bulky (about 20K at size), so interning them
          * to keep the size under control. When a programmatic client is not smart enough to
          * reuse a session, this helps keeping the memory consumption low.
          */
         private final LRUMap attributesCache = new LRUMap(32);
 
+        @Deprecated
         LDAPUserDetailsService(WebApplicationContext appContext) {
-            this(appContext, null);
+            this(appContext, null, null);
```

Comment:
Nope, it is this signature that I want to call `LDAPUserDetailsService(WebApplicationContext appContext, LDAPGroupMembershipStrategy groupMembershipStrategy, String server)`